spark/build.gradle

```diff
@@ -0,0 +1,63 @@
+import com.github.jengelman.gradle.plugins.shadow.transformers.ServiceFileTransformer
+
+plugins {
+  id 'application'
+  id 'java'
+  id 'com.diffplug.spotless' version '5.1.0'
+  id 'com.github.johnrengelman.shadow' version '6.0.0'
+}
+
+repositories {
+  mavenLocal()
+  mavenCentral()
+}
+
+ext {
+  lombokVersion = '1.18.12'
+}
+
+dependencies {
+  compileOnly 'org.apache.spark:spark-sql_2.11:2.4.6'
```

Comment:
Why not 3.0.0, since that's in the _Requirements_ section above?

Reply:
I moved the code from our internal repo, and didn't think to update the spark version 👍

Reply:
I get an ambiguous error for `filter()` with Spark 3+.  I'll change the requirement to 2.4+ and upgrade at a later time to spark 3+.

```
> Task :compileJava FAILED
/Users/wsl/dev/com.datakin/demo/spark/src/main/java/example/LinesWithAOrB.java:11: error: reference to filter is ambiguous
    final long numOfLinesWithAs = data.filter(line -> line.contains("a")).count();
```

Reply:
Is someone likely to follow this guide and spin up a Databricks cluster that has Spark 3 and run into an error? Or does the compile-time error not translate to a runtime error?

Reply:
You run into this issue when building the `jar`: `./gradlew shadowJar`. You can get around the issue by casting to the correct function class, see [this]( https://stackoverflow.com/questions/59712047/java8-maven-raise-error-reference-to-filter-is-ambiguous) thread

Reply:
Sure - so then a user shouldn't run into trouble running on Spark 3 because the compiled bytecode will have an explicit reference to the method being invoked. As long as the method signature is the same in both Spark 2 and Spark 3 everything should be fine.